### PR TITLE
Implement option for zonally asymmetric SST

### PR DESCRIPTION
**Pull request by KahaanGandhi**
## Purpose: 

Implementing a zonally asymmetric setup for sea surface temperature by perturbing a zonally symmetric setup and allowing users to specify their desired functional form.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/solver/types.jl [hunk 1]
```diff
@@ -18,6 +18,10 @@ struct SphericalModel <: AbstractModelConfig end
 struct BoxModel <: AbstractModelConfig end
 struct PlaneModel <: AbstractModelConfig end
 
+abstract type AbstractSST end
+struct ZonallySymmetric <: AbstractSST end
+struct ZonallyAsymmetric <: AbstractSST end
```
Comment (charleskawczynski) [inline src/solver/types.jl:23]:
```suggestion
struct ZonallySymmetricSST <: AbstractSST end
struct ZonallyAsymmetricSST <: AbstractSST end
```

File: src/solver/cli_options.jl [hunk 2]
```diff
@@ -94,6 +94,10 @@ function argparse_settings()
         help = "Surface thermo state type [`GCMSurfaceThermoState` (default), `PrescribedThermoState`]"
         arg_type = String
         default = "GCMSurfaceThermoState"
+        "--SST"
```
Comment (szy21) [inline src/solver/cli_options.jl:97]:
```suggestion
        "--surface_temperature"
```

File: src/solver/cli_options.jl [hunk 3]
```diff
@@ -94,6 +94,10 @@ function argparse_settings()
...
+        help = "Sea surface temperature scheme ['ZonallySymmetric' (default), 'ZonallyAsymmetric']"
```
Comment (szy21) [inline src/solver/cli_options.jl:98]:
```suggestion
        help = "Prescribed surface temperature functional form ['ZonallySymmetric' (default), 'ZonallyAsymmetric']"
```

File: src/solver/model_getters.jl [hunk 4]
```diff
@@ -24,6 +24,16 @@ function get_model_config(parsed_args)
     end
 end
 
+function get_sea_sfc_temp(parsed_args)
```
Comment (szy21) [inline src/solver/model_getters.jl:27]:
```suggestion
function get_sea_sfc_temp(parsed_args)
```
```suggestion
function get_sfc_temperature(parsed_args)
```
@charleskawczynski Any idea for a better name? This is not really getting the surface temperature, it is just getting the functional form (symmetric or asymmetric).

File: src/solver/type_getters.jl [hunk 5]
```diff
@@ -72,6 +72,7 @@ function get_atmos(config::AtmosConfig, turbconv_params)
         vert_diff,
         viscous_sponge = get_viscous_sponge_model(parsed_args, FT),
         rayleigh_sponge = get_rayleigh_sponge_model(parsed_args, FT),
+        sea_sfc_temp = get_sea_sfc_temp(parsed_args),
```
Comment (szy21) [inline src/solver/type_getters.jl:75]:
```suggestion
        sfc_temperature = get_sea_sfc_temp(parsed_args),
```

File: src/surface_conditions/surface_conditions.jl [hunk 6]
```diff
@@ -153,10 +153,37 @@ function surface_state_to_conditions(
...
+                T =
+                    (
+                        (-60 < coordinates.lat < 60) ?
+                        (
+                            FT(27) * (
+                                FT(1) -
+                                sinpi(
+                                    (FT(3) * coordinates.lat) /
+                                    (FT(2) * FT(180)),
+                                )^2
```
Comment (szy21) [inline src/surface_conditions/surface_conditions.jl:169]:
```suggestion
                                sind(
                                    (FT(3) * coordinates.lat) /
                                    FT(2),
                                )^2
```

File: src/surface_conditions/surface_conditions.jl [hunk 7]
```diff
@@ -153,10 +153,37 @@ function surface_state_to_conditions(
...
+                            ) + FT(273.16)
+                        ) : FT(273.16)
+                    ) + (
+                        (
+                            -180 < coordinates.long < 180 &&
+                            -30 < coordinates.lat < 30
+                        ) ?
+                        (
+                            FT(3) *
+                            cospi((coordinates.long + FT(90)) / (FT(180))) *
```
Comment (szy21) [inline src/surface_conditions/surface_conditions.jl:179]:
```suggestion
                            cosd(coordinates.long + FT(90)) *
```

File: src/solver/model_getters.jl (hunk 4 above)
Comment (charleskawczynski) [reply src/solver/model_getters.jl]: `sfc_temperature_form`?

Comment (szy21) [reply src/solver/model_getters.jl]: Sounds good to me, thanks!

File: .buildkite/pipeline.yml [hunk 8]
```diff
@@ -286,6 +276,14 @@ steps:
         agents:
           slurm_mem: 20GB
 
+      -label: "computer: aquaplanet (ρe_tot) equilmoist allsky radiation monin_obukhov varying insolation gravity wave (raw_topo) high top zonally asymmetric"
```
Comment (szy21) [inline .buildkite/pipeline.yml:279]:
```suggestion
      - label: "computer: aquaplanet (ρe_tot) equilmoist allsky radiation monin_obukhov varying insolation gravity wave (raw_topo) high top zonally asymmetric"
```

File: .buildkite/pipeline.yml [hunk 9]
```diff
@@ -286,6 +276,14 @@ steps:
...
+      command:
+        - julia --color=yes --project=examples examples/hybrid/driver.jl --z_max 45e3 --z_elem 25 --dz_bottom 300 --idealized_insolation false --vert_diff true --surface_setup DefaultMoninObukhov --moist equil --rad allskywithclear --non_orographic_gravity_wave true --orographic_gravity_wave raw_topo --precip_model 0M --rayleigh_sponge true --alpha_rayleigh_uh 0 --zd_rayleigh 30e3 --surface_temperature ZonallyAsymmetric --job_id sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --dt 400secs --t_end 18hours --dt_save_to_disk 18hours --regression_test true
+        - julia --color=yes --project=examples post_processing/remap/remap_pipeline.jl --data_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --out_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric
```
Comment (szy21) [inline .buildkite/pipeline.yml:282]:
```suggestion
        - julia --color=yes --project=examples post_processing/remap/remap_pipeline.jl --data_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --out_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric  
        - julia --color=yes --project=examples post_processing/plot/plot_pipeline.jl --nc_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --fig_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --case_name aquaplanet
```

File: src/solver/types.jl [hunk 10]
```diff
@@ -241,6 +246,7 @@ Base.@kwdef struct AtmosModel{
     vert_diff::VD = nothing
     viscous_sponge::VS = nothing
     rayleigh_sponge::RS = nothing
+    sfc_temperature::SST = nothing
```
Comment (szy21) [inline src/solver/types.jl:249]:
```suggestion
    sfc_temperature::ST = nothing
```

File: .buildkite/pipeline.yml [hunk 11]
```diff
@@ -286,6 +276,16 @@ steps:
...
+          - julia --color=yes --project=examples post_processing/remap/remap_pipeline.jl --data_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --out_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric
+          - julia --color=yes --project=examples post_processing/plot/plot_pipeline.jl --nc_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --fig_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --case_name aquaplanet
+          - julia --color=yes --project=examples regression_tests/test_mse.jl --job_id sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric --out_dir sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric
+        artifact paths: "sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric/*"
```
Comment (szy21) [inline .buildkite/pipeline.yml:285]:
```suggestion
        artifact_paths: "sphere_aquaplanet_rhoe_equilmoist_allsky_gw_raw_zonallyasymmetric/*"
```

Comment (szy21) [reply .buildkite/pipeline.yml:285]: see the error here: https://buildkite.com/clima/climaatmos-ci/builds/11289#018969fe-b306-4a4b-afc4-8b46c91a4442/150-157

**Comment by KahaanGandhi** [comment]
SST
A visualization of the zonally asymmetric functional form for surface temperature.

**Review by szy21 — APPROVED** [review]
Great work, thanks!

**Pull request merged**